src/backend/columnar/cstore_tableam.c

```diff
@@ -814,7 +815,7 @@ TruncateCStore(Relation rel, int elevel)
 		logical_to_smgr(GetHighestUsedAddress(rel->rd_node));
 
 	BlockNumber new_rel_pages = highestPhysicalAddress.blockno + 1;
-	if (new_rel_pages == old_rel_pages)
+	if (new_rel_pages >= old_rel_pages)
```

Comment:
For empty relation `highestPhysicalAddress.blockno` will be 0, so `new_rel_pages` will be 1 which is more than `old_rel_pages`.